### PR TITLE
Add uniform way to serialize containers of Expr to stream.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/expr/expr_template.h [hunk 1]
```diff
@@ -141,6 +144,65 @@ std::ostream& operator<<(std::ostream& out,
...
+ * Serialize a map of expressions to the given stream.
+ *
+ * @param out the output stream to use
+ * @param container the map of expressions to output to the stream
+ * @return the stream
+ */
+template <typename V>
+std::ostream& operator<<(std::ostream& out, const std::map<Expr, V>& container)
+{
+  container_to_stream(out, container);
```
Comment (4tXJ7f) [inline src/expr/expr_template.h]: The definition of this function can be moved to the source file like the other variants.

File: src/expr/expr_template.h [hunk 2]
```diff
@@ -141,6 +144,65 @@ std::ostream& operator<<(std::ostream& out,
...
+ * @param out the output stream to use
+ * @param container the unordered_map of expressions to output to the stream
+ * @return the stream
+ */
+template <typename V>
+std::ostream& operator<<(
+    std::ostream& out,
+    const std::unordered_map<Expr, V, ExprHashFunction>& container)
+{
+  container_to_stream(out, container);
```
Comment (4tXJ7f) [inline src/expr/expr_template.h]: The definition of this function can be moved to the source file like the other variants.

File: src/expr/expr_template.h (hunk 1 above)
Comment (aniemetz) [reply src/expr/expr_template.h]: Done

File: src/expr/expr_template.h (hunk 2 above)
Comment (aniemetz) [reply src/expr/expr_template.h]: Done

**Pull request merged**